controllers/kuadrant_controller.go

```diff
@@ -181,44 +181,40 @@ func (r *KuadrantReconciler) unregisterExternalAuthorizer(ctx context.Context) e
 
 func (r *KuadrantReconciler) unregisterExternalAuthorizerIstio(ctx context.Context) error {
 	logger, _ := logr.FromContext(ctx)
-	iop := &iopv1alpha1.IstioOperator{}
+	var configsToUpdate []istio.ConfigWrapper
 
+	iop := &iopv1alpha1.IstioOperator{}
 	iopKey := client.ObjectKey{Name: controlPlaneProviderName(), Namespace: controlPlaneProviderNamespace()}
 	if err := r.Client().Get(ctx, iopKey, iop); err != nil {
 		logger.V(1).Info("failed to get istiooperator object", "key", iopKey, "err", err)
-		return err
+		if apimeta.IsNoMatchError(err) {
+			// return err only if there's no istiooperator CRD
+			return err
+		}
+		// otherwise, we assume that the control plane is istio but no installed by its operator
```

Comment:
IDK if I understand returning only if IsNoMatchError 🤔  This will cause the reconciliation to try to proceed before other API errors, such as a 429 (occasionally raised by the [API Priority and Fairness](https://kubernetes.io/docs/concepts/cluster-administration/flow-control/) filter). Do we really want this?